File: src/main/java/com/nkming/powermenu/PersistentViewHelper.java
/*
 * PersistentViewHelper.java
 *
 * Author: Ming Tsang
 * Copyright (c) 2015 Ming Tsang
 * Refer to LICENSE for details
 */

package com.nkming.powermenu;

import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;

public class PersistentViewHelper
{
	public static void startIfNecessary(Context context)
	{
		SharedPreferences pref = context.getSharedPreferences(context.getString(
				R.string.pref_file), Context.MODE_PRIVATE);
		if (pref.getBoolean(context.getString(R.string.pref_persistent_view_key),
				false))
		{
			Log.d(LOG_TAG + ".startIfNecessary", "Starting service");
			context.startService(new Intent(context, PersistentService.class));
		}
	}

	private static final String LOG_TAG =
			PersistentViewHelper.class.getCanonicalName();
}

File: src/main/kotlin/com/nkming/powermenu/PersistentViewHelper.kt
package com.nkming.powermenu

import android.content.Context

object PersistentViewHelper
{
	@JvmStatic
	fun startIfNecessary(context: Context): Boolean
	{
		val pref = Preference.from(context)
		if (pref.isPersistentViewEnabled
				&& !com.nkming.utils.widget.PersistentService.isRunning())
		{
			if (!ensurePermission(context))
			{
				return false
			}
			else
			{
				Log.d("$LOG_TAG.startIfNecessary", "Starting service")
				PersistentService.start(context)
				return true
			}
		}
		return false
	}

	private val LOG_TAG = PersistentViewHelper::class.java.canonicalName

	private fun ensurePermission(context: Context): Boolean
	{
		if (PermissionUtils.hasSystemAlertWindow(context))
		{
			return true
		}
		else
		{
			Log.d("$LOG_TAG.ensurePermission", "Missing overlay permission")
			PermissionUtils.requestSystemAlertWindow(context)
			return false
		}
	}
}
